wallet-new/server/Cardano/Wallet/API/V1/Handlers/Addresses.hs

```diff
@@ -31,5 +39,11 @@ listAddresses RequestParams {..} = do
                       }
             }
 
-newAddress :: Address -> Handler (WalletResponse Address)
-newAddress a = return $ single a
+newAddress
+    :: (MonadThrow m, V0.MonadWalletLogic ctx m)
+    => NewAddress -> m WalletAddress
```

Comment:
This is easy to forget (maybe we should enforce this at the type level) but we now need to send back `(WalletResponse WalletAddress)` so that we get the nice JSend format with pagination.

This means you need to wrap the final line into `single <$> ...`